app/services/SMService.scala

```diff
@@ -93,14 +92,21 @@ trait SMService extends Logging {
     jsonConnector.loadServicesJson.fields map { case (service, _) => service}
   }
 
-  def searchForService(query: String): Seq[String]= {
-    if(query.contains("\"")) {
-      exactMatchSearch(jsonConnector.loadServicesJson.fields, query)
-    } else {
-      jsonConnector.loadServicesJson.fields.collect { case (name, _) if name.contains(query) => name}
+  def searchForService(query: String): Seq[String] = {
+    val servicesJson: Seq[(String, JsValue)] = jsonConnector.loadServicesJson.fields
+
+    (isAnInt(query), query.contains("\"")) match {
+      case (true, _) => searchServicesUsingPort(query, servicesJson)
+      case (_, true) => exactMatchSearch(servicesJson, query.toUpperCase)
+      case _ => servicesJson.collect { case (name, _) if name.contains(query.toUpperCase) => name }
     }
   }
 
+  private def isAnInt(query: String): Boolean = Try(query.toInt).isSuccess
+
+  private def searchServicesUsingPort(query: String, servicesJson: => Seq[(String, JsValue)]): Seq[String] =
+    servicesJson.collect { case (name, js) if js.\("defaultPort").as[Int].toString.contains(query) => name }
```

Comment:
Where you get the default port and use `as[Int]`, i've found it is failing for me as it's trying to process `MONGO` as a service. Unfortunately mongo doesn't have a port defined. Switching the `as[Int]` to `asOpt[Int]` should work